### ISSUE TITLE
AR sampling with KV Caching

### ISSUE THREAD

**Issue by Tejomay128** [ML-GSAI/SMDM#5]
I had a question about the implementation of the AR sampling with the KV cache implemented in the `gen_model_answer.py` file. There, the generation loop seems to terminate when the first element of the batch samples an EOS token (in line no. 70 `if next_token[0] == torch.tensor([tokenizer.eos_token_id], device=device):`). However, it might lead to an early termination for other elements in the batch during generation. Am I missing something?

Also, the padding is on the right side during AR generation as well. Usually what I have seen is that we left pad the prompts and then generate.
Would like some clarification on this.

**Comment by nieshenx** [comment]
Thank you for your question! When sampling, I set the batch size to 1, so my setting is correct. However, this is indeed a very rudimentary implementation.

**Comment by Tejomay128** [comment]
Understood. Thanks for the clarification.
